internal/controller/suite_test.go

```diff
@@ -864,16 +866,58 @@ func randomString(length int) string {
 	return string(result)
 }
 
-func simulateWebhook(ctx context.Context, k8sClient client.Client, ctp *promoterv1alpha1.ChangeTransferPolicy) {
-	Eventually(func(g Gomega) {
-		orig := ctp.DeepCopy()
-		if ctp.Annotations == nil {
-			ctp.Annotations = make(map[string]string)
-		}
-		ctp.Annotations[promoterv1alpha1.ReconcileAtAnnotation] = metav1.Now().Format(time.RFC3339)
-		err := k8sClient.Patch(ctx, ctp, client.MergeFrom(orig))
-		Expect(err).To(Succeed())
-	}, constants.EventuallyTimeout).Should(Succeed())
+// buildGitHubWebhookPayload constructs a GitHub webhook payload for push events
+func buildGitHubWebhookPayload(beforeSha, ref string) string {
+	payload := map[string]any{
+		"before": beforeSha,
+		"ref":    ref,
+		"pusher": map[string]any{
+			"name":  "test-user",
+			"email": "test@example.com",
+		},
+	}
+	payloadBytes, err := json.Marshal(payload)
+	Expect(err).NotTo(HaveOccurred())
+	return string(payloadBytes)
+}
+
+// triggerWebhook triggers a CTP reconciliation by sending an actual webhook request to the webhook receiver server.
+// If the CTP doesn't have a proposed SHA yet (e.g., on first reconciliation), it falls back to annotation-based triggering.
+func triggerWebhook(ctx context.Context, k8sClient client.Client, ctp *promoterv1alpha1.ChangeTransferPolicy) {
+	// Refresh the CTP to get the latest status with the proposed SHA
+	err := k8sClient.Get(ctx, types.NamespacedName{
+		Name:      ctp.Name,
+		Namespace: ctp.Namespace,
+	}, ctp)
+	Expect(err).To(Succeed())
+
+	// Get the SHA from the CTP status that the webhook receiver will look for
```

Comment:
```suggestion
	// Get the SHA from the CTP status that the webhook receiver will look for
	// TODO: get this from git instead of the CTP to better reflect real git behavior
```

Is this basically true? This would let us test scenarios where the CTP basically gets out of date vs. what's in git.